### ISSUE TITLE
Terrible performance when using a jpeg in combination with node.js

### ISSUE THREAD

**Issue by bartvanremortele** [naptha/tesseract.js#53]
I seem to be having performance issues when using tesseract.js on node in combination with a jpeg. When I run the basic.js example using npm installed modules instead of the local modules I get pretty good results: "Benchmark took 3034.437842 miliseconds"

However as soon as I run the same example using a jpeg of 912 × 2121 px the results are very poor: 
"Benchmark took 41944.719605 miliseconds"

If I run the same jpeg image in the browser example it gives the same performance-ish as the .png on node. I think it has something to do with the way ~~loadImage~~ is implemented in the node.js version but I haven been able to pin it down.

**Edit:** The delay is past the loading of the image itself and is probably related to the interprocess communication. The decoded raw array that is being sent is huge. This doesn't cause a delay in the browser however.

**Comment by bijection** [comment]
Interesting! Is the raw array smaller for a png of the same size?

**Comment by rowasc** [comment]
Posting findings in case it helps someone debug this issue. It'd be great to have a couple of sets of images to try this out

### Medium size image (960px x 720px)
Execution time from start to finish

**Tesseract JPG (22Kb)**
- 7836.192ms
- 7683.145ms
- 7327.576ms
- 7546.628ms

**Tesseract PNG (11kb)**
- 9132.705ms
- 9054.147ms
- 8672.093ms
- 9110.318ms

This is the set of images I used (dummies I created with google draw.. )

![jpgpng-tesseract-2](https://cloud.githubusercontent.com/assets/2434401/19460215/f683345c-94af-11e6-9a49-60f7ef833f12.png)
![jpgpng-tesseract-2](https://cloud.githubusercontent.com/assets/2434401/19460217/f85515e8-94af-11e6-9c57-a382bd443f59.jpg)

I'm running a second set of tests with bigger images (>100kb) that looks like it's going to yield pretty much the same results (>2 minutes processing time for each file on the first run, with the 177kb JPG taking ~13 second less than the 100Kb PNG)... weird.

**Comment by rowasc** [comment]
![jpgpng-bigfile](https://cloud.githubusercontent.com/assets/2434401/19460367/524d8a84-94b1-11e6-88d8-2ac33a094ac2.png)
![jpgpng-bigfile](https://cloud.githubusercontent.com/assets/2434401/19460368/52868654-94b1-11e6-8091-c8ee8bb17551.jpg)

**Comment by bartvanremortele** [comment]
@rowasc I can provide you with some testing images:

![3_processed](https://cloud.githubusercontent.com/assets/424370/19468686/10d2c44a-9518-11e6-9068-31937b7620d7.jpg)
![2_processed](https://cloud.githubusercontent.com/assets/424370/19468687/10d67784-9518-11e6-9048-c2ea25b8c3bd.jpg)
![1_processed 1](https://cloud.githubusercontent.com/assets/424370/19468685/10d2636a-9518-11e6-8f07-50f78839635a.jpg)

**Comment by rowasc** [comment]
Thanks ! yes, that makes total sense. 
I'll get back to  this after work today.

**Comment by bartvanremortele** [comment]
@rowasc Have you looked into the performance yet? I'm interested in further contribution if I can help.

**Comment by pabx06** [comment]
bmp image  width in pixel:608	 height in pixel:300
TEXT is a THREE LETTTER UPERCASE sans seriff. 

i get TIME spent (ms)= 5330, TEXT= OAX,  CONFIDENCE 92

way too slow compared to human.
i am wondering if it has something todo with preloading of language file ?

how can i tune it up for many small stream of images ?

**Comment by jeromewu** [comment]
Please use the latest version and see if the performance issue still exists, feel free to open new issues, close this one for now.

**Comment by GrillAndChill** [comment]
@jeromewu I used the new version, but it still takes almost a minute for images with more than 50 words.

So I have a POST endpoint called "/image2text". 
- I have tried creating a worker and loading the lang data ahead of time (out side of the POST api call), and
- I have also tried keeping it all together inside the POST call (creating the worker and loading the lang data once the user make a POST call, then terrminate the worker after the call has been fulfilled). 

However, the results are the same. I didn't notice any improvement in the performance.
